particle-iot/device-os
# Update the Tinker firmware to support configuring Muon.

### Problem
The factory default user app, i.e. the Tinker firmware, running on Particle M.2 SoMs that is installed on Muon may not work as expected, since some of the pins for controlling the power module and Etehrnet are not correct.

### Solution
Introduce two USB requests to
1. identify whether the base board is Muon or not
2. set correct control pins for PMIC and Ethernet.

### Steps to Test
1. Build and flash the tinker app to a Particle M.2 SoM
2. Update the `DVOS/user/tests/app/usb_ctrl_request/test/test.js` with the following content
3. Follow the `DVOS/user/tests/app/usb_ctrl_request/README.md` to run the test
```js
import * as control from './support/control';
import * as proto from './support/proto';
import * as usb from './support/usb';
import './support/init';

import * as semver from 'semver';
import * as randomstring from 'randomstring';

import { expect } from 'chai';

describe('USB requests', function() {
  let dev = null;

  before('find a Particle USB device', function() {
    return usb.enumDevices().then((devs) => {
      if (devs.length == 0) {
        throw new Error('No Particle USB devices found');
      }
      dev = devs[0]; // Use the first device in the list
    });
  });

  before('open device', function() {
    return dev.open();
  });

  after('close device', function() {
    if (dev) {
      return dev.close();
    }
  });

  describe('application control requests', function() {
    this.slow(2000); // Test application replies with a random delay

    it('identify base board should succeed', function() {
      const reqData = '{"BASE_BOARD_DETECT":null}';
      console.log(reqData);
      return control.sendRequest(dev, control.RequestType.APP_CUSTOM, reqData).should.become({
        result: control.Result.OK,
        data: '{"board":"muon"}'
      });
    });

    it('configure base board should succeed', function() {
        const reqData = '{"CONFIGURE_MODULE_BOARD":"muon"}';
        console.log(reqData);
        return control.sendRequest(dev, control.RequestType.APP_CUSTOM, reqData).should.become({
          result: control.Result.OK,
          data: '{"status":0}'
        });
      });
  });
});
```

### Example App
`make PLATFORM=msom all APP=tinker`

### References
N/A

---

### Completeness
